### ISSUE TITLE
torch.svd allocates too large a matrix with 'S'

### ISSUE THREAD

**Issue by soumith** [torch/torch7#227]
torch.svd allocates too large a matrix in the setting where you take the svd of an mxn matrix A with n>m, independent of  whether the 'S' flag is used.  If it is going to return an mxm matrix for the right singular vectors (instead of an mxn "economy sized" matrix), it should return an o.n. basis for R^m, with the first n vectors in the basis spanning the range of  A; currently the last m-n columns of the right singular value matrix are junk...

Reported by @aszlam

**Comment by yf225** [comment]
Instead of allocating an m x m matrix, I propose that we should return an m x n matrix by tossing out the last m-n columns of the original matrix. This way we can save quite a bit of memory, since the junk values are useless anyways. 

Thoughts? I can work on this if the change makes sense.

**Comment by soumith** [comment]
cc: @koraykv @dominikgrewe we're proposing this change. Are you okay with this? what are your thoughts.

**Comment by soumith** [comment]
ping ping @koraykv @dominikgrewe

**Comment by dominikgrewe** [comment]
Sorry for the delay. I'm not very familiar with svd but @koraykv said he wanted to have a look.

**Comment by koraykv** [comment]
Yes, in the current implementation the flags for fat matrices are not implemented, only tall case is considered. Sorry for taking long.
